storagetest/storagetest.go

```diff
@@ -2,69 +2,232 @@ package storagetest
 
 import (
 	"context"
+	"fmt"
+	"path"
+	"sort"
+	"strings"
+	"sync/atomic"
 	"testing"
 
 	"github.com/giantswarm/microstorage"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 )
 
 // Test is Storage conformance test.
 func Test(t *testing.T, storage microstorage.Storage) {
```

Comment:
You just call the suite from implementation test function like: https://github.com/giantswarm/microstorage/blob/master/memory/storage_test.go#L69-L75

I tried to extract a few tests that do one thing. It's impossible to do it in a "pure" fashion because you have to rely on other functionalities. For example on `Search` when you test `Create`. I did my best though.

There is a call to several private methods instead of exposing bunch of public ones, because if the suit is further extended it's difficult to enforce update on all the implementations. But if we add another private method call here it will happen eventually with vendor updates.

I'm uncertain if this question is about how to execute those test, or complaint about them being complicated. I hope I answered it. If no LMK.